### ISSUE TITLE
Galleries are no longer visible for admin

### ISSUE THREAD

**Issue by humantex** [stfnhmplr/pagekit-gallery#40]
I've just upgraded from 0.2.6 to 0.3.0 and lost the entire list of 4 galleries that were there minutes before. Even worse news is that the front-end of the site is totally trashed for the individual gallery pages, the home page, and the blog pages - with the following error showing via debug:

`FatalErrorException in ImageModelTrait.php line 59: Error: Call to a member function cropResize() on null`
... or ...
when trying to open a gallery page: `FatalErrorException in ImageModelTrait.php line 38: Error: Call to a member function zoomCrop() on null` 

I tried the fix from the #29 closed issue, and modifying all of the toggles and relevant default settings, but still no listings showing, and no change to the site overall.

The only linked pages I get without errors are a FormMaker 'Contact us' page, the FriendlyIT 'Search' page, and ironically, the overall 'Galleries' page. Nothing else will render correctly without some form of error.

There are additional gallery-related console errors from FireFox devTools when I open the contact-us page:
```
[vue-loader] app/components/link-gallery.vue: named exports in *.vue files are ignored.
link-gallery.js:52:4
[vue-loader] app/components/gallery-preview.vue: named exports in *.vue files are ignored.
editor-plugin.js:147:4
[vue-loader] app/components/gallery-picker.vue: named exports in *.vue files are ignored.
editor-plugin.js:208:4
Loading failed for the <script> with source “sitemap:app/bundle/link.js”.
contact-us:43
```
With this being a site about photography, having no gallery is the same as having no site at all. Help would be greatly appreciated.

Pagekit - v.1.0.13
Web Server - LiteSpeed
PHP - 5.6.32
DB - sqlite v.3.8.5
gallery ext. - v.0.3.0

Site URL: https://enami.lumiworx.com

**Comment by humantex** [comment]
I'm not sure how relevant these observations will be, but if it helps at all...

Line 37 of ImageModelTrait.php shows a path setting for the cache directory, which is one line above the call for 'zoomCrop:
`->setCacheDir('storage/shw-gallery/cache')`

A similar line at 58 - one notch above the call for cropResize - reads:
`->setCacheDir('storage/shw-gallery/cache')`

There is no such directory in my site tree... I created one, but no change whatsoever.

Looking at the DB tables, I don't see any entries labeled as 'width' or 'height' or anything related to zoom/crop/sizing or cache within either the "_gallery" table, or the "_images" table. Cropped screen-grab attached...

![pagekit-gallery-schema](https://user-images.githubusercontent.com/16087649/32739974-72b4d6ae-c86f-11e7-9fb7-23085152adec.jpg)

**Comment by humantex** [comment]
One last bit of info from the debug toolbar... there were 7 DB calls when opening the initial gallery list page, and the last 2 calls appear to have returned no data - i.e.,:

```
SELECT user_id, name, username FROM ls_galleries INNER JOIN ls_system_user ON user_id = ls_system_user.id GROUP BY user_id, name

101μs {}

SELECT * FROM ls_system_role WHERE id IN (2,3)

149μs {}

```

At this point I'm out of ideas. I do have a backup, of sorts, that I can upload to the server that should (I assume) get me back to a running site with gallery v.0.2.6.

It will be a royal pain to do, since I'll have to scrape 30 hours worth of changes from a copy of the DB as it was last filled with new content, but it may have to suffice as a stop-gap measure until I can find a suitable replacement for pagekit, et-all. I'm not well versed in how pagekit works to be able to fork and fix.

**Comment by stfnhmplr** [comment]
Thanks for your feedback. The issue was a custom directory for image caching. Should be fixed with 0.3.1.

**Comment by humantex** [comment]
That was amazingly fast and it fixed everything within seconds. Outstanding! Thank you for your help.

Again... **absolutely outstanding!**
